core/src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala

```diff
@@ -432,6 +433,59 @@ object CSVScriptSignature extends ScriptFactory[CSVScriptSignature] {
   }
 }
 
+sealed trait ConditionalScriptSignature extends ScriptSignature {
+  require(asm.lastOption.contains(OP_TRUE) || asm.lastOption.contains(OP_FALSE),
+          "ConditionalScriptSignature must end in true or false")
+
+  def isTrue: Boolean = {
+    asm.last == OP_TRUE
```

Comment:
It would be nice to have a `BitcoinScriptUtil.isTrue()` and `BitcoinScriptUtil.isFalse()`, note we do do have a [`BitcoinScriptUtil.castToBool()`](https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/util/BitcoinScriptUtil.scala#L516).

Sadly something being considered a bool isn't as strict as `OP_TRUE/OP_1/OP_0/OP_FALSE` which is why we have a policy flag for `MINIMALIF`

Reply:
I'm not sure why we would ever use a `BitcoinScriptUtil.isTrue` as opposed to `== OP_TRUE`? Could you elaborate? (note that in this specific case it will certainly be either `OP_TRUE` or `OP_FALSE` by the invariant)